File: java/src/main/java/br/com/atlas/averbacao/api/contratacao/ContratacaoController.java
package br.com.atlas.averbacao.api.contratacao;

import br.com.atlas.averbacao.services.contratacao.ContratacaoContract;
import io.swagger.v3.oas.annotations.Operation;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class ContratacaoController {
    private final ContratacaoContract contratacao;

    public ContratacaoController(ContratacaoContract contratacao) {
        this.contratacao = contratacao;
    }

    @Operation(summary = "Uma oferta sua foi escolhida — confirme no seu core")
    @PostMapping("/v1/contratacoes")
    public OkResponse post(@Valid @RequestBody ContratacaoRequest request) {
        contratacao.iniciada(request);
        return new OkResponse();
    }
}

File: dotnet/Atlas.Averbacao/Api/Contratacao/ContratacaoController.cs
using Atlas.Averbacao.Api.Contratacao;
using Atlas.Averbacao.Services.Contratacao;
using Microsoft.AspNetCore.Mvc;

namespace Atlas.Averbacao.Api.Contratacao;

[ApiController]
[Route("v1/contratacoes")]
public sealed class ContratacaoController(IContratacaoService contratacao) : ControllerBase
{
    /// <summary>Uma oferta sua foi escolhida — confirme no seu core.</summary>
    [HttpPost]
    [ProducesResponseType(typeof(OkResponse), 200)]
    public async Task<OkResponse> Post([FromBody] ContratacaoRequest request)
    {
        await contratacao.Iniciada(request);
        return new OkResponse();
    }
}
